tutorials/grid_coordinates.py

```diff
@@ -0,0 +1,346 @@
+"""
+.. _grid_coordinates:
+
+Gridding coordinates
+====================
+
+Grid coordinates in Verde are used to create points on a regularly spaced grid
+that are then used in the spline method to interpolate between sample data
+points. The grid can be specified either by the number of points in each
+dimension (the *shape*) or by the grid node spacing.
+
+Creating regular grids in Verde is done using the :func:`verde.grid_coordinates`
+function. It creates a set of regularly spaced points in both the west-east and
+south-north directions, i.e. a two-dimensional spatial grid.
+
+First let's create a region that is 1000 units west-east and 1000 units south-
+north, and we will set an initial spacing to 100 units.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import verde as vd
+from matplotlib.patches import Rectangle
+
+# create the bounds for the region and set the spacing for the grid points
+spacing = 100
+west, east, south, north = 0, 1000, 0, 1000
+region = (west, east, south, north)
+
+# create the grid
+square_region_east, square_region_north = vd.grid_coordinates(
+    region=region, spacing=spacing
+)
+
+########################################################################################
+# We can check the dimensions of the grid to confirm that it is 11x11
+
+print(square_region_east.shape, square_region_north.shape)
+
+########################################################################################
+# Now let's plot the grid to visualize the location of the grid points
+
+plt.figure(figsize=(6, 6))
+currentAxis = plt.gca()
+currentAxis.add_patch(
+    plt.Rectangle((west, south), east, north, fill=None, label="Region Bounds")
+)
+
+plt.vlines(
+    square_region_east[0],
+    ymin=min(square_region_north[0]),
+    ymax=max(square_region_north[-1]),
+    linestyles="dotted",
+    zorder=0,
+)
+plt.hlines(
+    square_region_north[:, 1],
+    xmin=min(square_region_east[0]),
+    xmax=max(square_region_east[-1]),
+    linestyles="dotted",
+    zorder=0,
+)
+
+plt.scatter(
+    square_region_east,
+    square_region_north,
+    label="Square Region Grid Nodes",
+    marker=".",
+    color="black",
+    s=100,
+)
+
+plt.ylim(-50, 1050)
+plt.xlim(-50, 1050)
+plt.xlabel("Easting")
+plt.ylabel("Northing")
+plt.legend(loc="upper center", bbox_to_anchor=(0.5, 1.15))
+plt.show()
+
+########################################################################################
+# Adjusting region boundaries when creating the grid
+# --------------------------------------------------
+# Now let's create a region that is 1555 units west-east, and 1250 units south-north

```

Comment:
It might be better to keep the region the same as the previous example and modify the spacing slightly so that it doesn't divide exactly anymore. This way, between the last example and this one only 1 thing changes (keeping the cognitive burden low).